### PR TITLE
W-14642249: Improve flow extension model to communicate min value for maxConcurrency parameter model

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: integration/src/test/java/org/mule/test/routing/AsyncTestCase.java [hunk 1]
```diff
@@ -95,7 +99,7 @@ public void psThreadingPropagated() throws Exception {
   }
 
   @Test
-  @Story(BACKPRESSURE)
+  @Stories({@Story(BACKPRESSURE), @Story(ExecutionEngineStory.MAX_CONCURRENCY)})
```
Comment (rbourbonnavarro) [inline integration/src/test/java/org/mule/test/routing/AsyncTestCase.java:102]: Static import.

File: integration/src/test/java/org/mule/test/routing/AsyncTestCase.java [hunk 2]
```diff
@@ -115,14 +119,14 @@ public void withMaxConcurrencyAsyncDispatched() throws Exception {
...
```
Comment (rbourbonnavarro) [inline integration/src/test/java/org/mule/test/routing/AsyncTestCase.java:122]: Static import.

File: integration/src/test/java/org/mule/test/routing/AsyncTestCase.java [hunk 3]
```diff
@@ -115,14 +119,14 @@ public void withMaxConcurrencyAsyncDispatched() throws Exception {
...
   @Description("Assert that if no maxConcurrency is configured for an async, the value from the flow is inherited")
   public void withFlowMaxConcurrency() throws Exception {
     testAsyncMaxConcurrency("with-flow-max-concurrency");
   }
 
   @Test
-  @Story(BACKPRESSURE)
+  @Stories({@Story(BACKPRESSURE), @Story(ExecutionEngineStory.MAX_CONCURRENCY)})
```
Comment (rbourbonnavarro) [inline integration/src/test/java/org/mule/test/routing/AsyncTestCase.java:129]: Static import.

**Review by rbourbonnavarro — APPROVED** [review]

File: integration/src/test/java/org/mule/test/routing/AsyncTestCase.java (hunk 3 above)
Comment (elrodro83) [reply integration/src/test/java/org/mule/test/routing/AsyncTestCase.java:129]: cannot, there's already a constant with that identifier

File: integration/src/test/java/org/mule/test/routing/AsyncTestCase.java (hunk 2 above)
Comment (elrodro83) [reply integration/src/test/java/org/mule/test/routing/AsyncTestCase.java:122]: cannot, there's already a constant with that identifier

File: integration/src/test/java/org/mule/test/routing/AsyncTestCase.java (hunk 1 above)
Comment (elrodro83) [reply integration/src/test/java/org/mule/test/routing/AsyncTestCase.java:102]: cannot, there's already a constant with that identifier

**Pull request merged**